src/plugins/proxy/tests/properties_tests.cpp

```diff
@@ -147,5 +305,41 @@ TEST_F(ProxyTests, get_property_for_changed_default_device) {
         }
     }
     EXPECT_EQ(5, immutable_pr);
-    EXPECT_EQ(2, mutable_pr);
+    EXPECT_EQ(3, mutable_pr);
 }
+
+#ifdef ENABLE_AUTO_BATCH
+
+TEST_F(ProxyTests, set_cache_dir_for_auto_batch__hetero_fallback_device) {
+    core.get_available_devices();
+    const std::string dev_name = "MOCK.1";
+    core.set_property(dev_name, ov::cache_dir("./test_cache"));
+    auto model = create_model_with_subtract();
+    auto compiled_model = core.compile_model(model,
+                                             "MOCK.1",
+                                             ov::cache_dir("./test_cache"),
+                                             ov::hint::performance_mode(ov::hint::PerformanceMode::THROUGHPUT));
+    auto infer_request = compiled_model.create_infer_request();
+    auto input_tensor = create_and_fill_tensor(model->input().get_element_type(), model->input().get_shape());
+    infer_request.set_input_tensor(input_tensor);
+    infer_request.infer();
+    CommonTestUtils::removeDir("./test_cache");
```

Comment:
Better to move folder maintenance in `SetUp`/`TearDown` or something like that (temp directory object with RAII around directory creation/deletion) because any exception in a code above won't trigger folder clean up, and next tests may be (and often will be) affected